pycontrails/models/cocipgrid/cocip_time_handling.py

```diff
@@ -340,3 +348,29 @@ class CocipRuntimeStats:
     n_old_vectors: int
     n_steps_new_vectors: int
     n_steps_old_vectors: int
+
+
+def _check_start_time(
+    met_start: pd.Timestap, model_start: pd.TimeStamp, name: str, note: str | None = None
+) -> None:
+    if met_start > model_start:
+        note = f" Note: {note}" if note else ""
+        warnings.warn(
+            f"Start time of parameter '{name}' ({met_start}) "
+            f"is after model start time ({model_start}). "
+            f"Include additional time at the start of '{name}'."
+            f"{note}"
+        )
+
+
+def _check_end_time(
+    met_end: pd.Timestap, model_end: pd.TimeStamp, name: str, note: str | None = None
```

Comment:
You may want to make `note` a keyword-only variable both here and in `_check_end_time`

`... name: str, note: str | None ...` -> `... name: str, *, note: str | None ...`